### DISCUSSION TITLE
Adding in-vehicle display

### DISCUSSION THREAD

**Discussion by tian96-pu** [cyberbotics/webots#5264] [Q&A]
Is there a way to add an in-vehicle display showing the location of other vehicle at webots simulation?

**Comment by omichel** [comment]
Certainly. You should equip the vehicle with a [Display](https://www.cyberbotics.com/doc/reference/display) device controlled by a supervisor process that tracks down the location of all vehicles and displays them here.

**Reply by tian96-pu** [reply]
Thanks for you reply. It is really helpful. Could you privide more informaiton about how to track the vehicles locations? Do I need to add more devices or is there any way I can get vehicle information from SUMO supervisor or somewhere else?
I am using the SUMO supervisor to generate the background vehicles and using the webots vehicles as the ego vehicles.

**Reply by omichel** [reply]
I guess you can edit the source code of the SUMO supervisor to get the position of the vehicles.

**Reply by tian96-pu** [reply]
Thanks omichel. That worked. I have already added the dymanic display on my vehicle. 
Could I aslo add dynamic display on the cautionpanel https://cyberbotics.com/doc/guide/object-traffic#cautionpanel?

**Reply by omichel** [reply]
>  Could I aslo add dynamic display on the cautionpanel https://cyberbotics.com/doc/guide/object-traffic#cautionpanel?

Yes, but you should probably create your own copy of the CautionPanel proto and modify it: turn it into a Robot and add a Display and a controller in it.

**Reply by tian96-pu** [reply]
Thanks! That works with creating my own CautionPanel.

**Reply by tian96-pu** [reply]
Hi olivier, after I use display.draw.text function to show the text on the display, how could I clear the disiplay with no text? I did not find the related function for display. 
Also, my vehicle reach a specific point, I would like to move the vehilce to another location. How could move the viewpoint to the new location as well?

**Reply by omichel** [reply]
To clear the display, you can simply [draw a filled rectangle](https://www.cyberbotics.com/doc/reference/display#wb_display_fill_rectangle) with the background color. You the use the supervisor API to move any node from the scene tree, see https://www.cyberbotics.com/doc/guide/supervisor-programming#setting-the-position-of-robots for example.

**Reply by tian96-pu** [reply]
Hi Michel, I would like to show the current speed of the vehicle on a display. For each simulation timestep, I used drawText to show the speed and the use the fillrectangle with the background color. At the next timestep, after I used the drawText, the display did not show any text. It seems the display was filled by the rectangle. Do you have any suggestions about how to design a dynamic display to show current speed of the vehicle?

**Reply by omichel** [reply]
Did you change the foreground color in between?

**Reply by tian96-pu** [reply]
The foreground color is not changes.

**Reply by omichel** [reply]
You should change it so that your text is visible on the filled rectangle.

**Reply by tian96-pu** [reply]
Hi Michel, is there a way to set the webots vehciles as automatic transmission?

**Reply by omichel** [reply]
Yes, `Car.gearRatio [-1 1]` will make the car automatic transmission. See for example the TestlaModel3 proto example.
